### ISSUE TITLE
Unable to reproduce anti-distillation results in paper

### ISSUE THREAD

**Issue by AknA13** [UNITES-Lab/DOGe#6]
Hi DOGe authors,

We've been trying to reproduce the results stated in the paper using the official repo, but have been having trouble reproducing the results.# Trial 1: Running the code as it is. This is the command line we've tried: bash generate.sh and bash train-doge.sh. After running these two, the responses from the fine-tuned DOGe teacher output text in a different language and repeated sequences due to the collapse when training. We found that the reasoning-aware masking approach mentioned in the paper was not implemented or used in the codebase, which might be causing this result mismatch. So when I tried training DOGe, the answers would be extremely manipulated because the KL loss is applied to both the answer and reasoning trace.# Trial 2: Running the code with reasoning-aware masking implemented. Therefore, I manually implemented the reasoning-aware masking myself by just applying the KL loss on the tokens in the <think> </think> boundaries.  I used the default KL coefficient specified in the paper (3e-5), and the loss curve looked different from the paper and didn't show much of an increase in adversarial loss. I tried using a KL COEF of 1e-2 and 1e-3, whose loss curve is also shown below. Loss curves are more stable with an increase in the adversarial loss and the SFT loss staying around the same/lower, as you can see in the attached graphs.

<img width="5056" height="2656" alt="Image" src="https://github.com/user-attachments/assets/a58af2db-391b-4f20-8e89-2e962600a756" />

<img width="5056" height="2656" alt="Image" src="https://github.com/user-attachments/assets/36c9ff70-2411-4964-9703-5c33f36967e9" />

<img width="5056" height="2656" alt="Image" src="https://github.com/user-attachments/assets/d8eb6af4-4c48-4ed8-bf33-aba4e04f9ae5" />

3.  And here are the evaluation results:

<img width="1890" height="948" alt="Image" src="https://github.com/user-attachments/assets/d1173ed1-ff15-4169-9735-19ba0e423149" />

<img width="1770" height="980" alt="Image" src="https://github.com/user-attachments/assets/c7311c79-5370-46a1-93da-36a9eafc6507" />

The paper states a very small drop in performance from the base teacher to the defensive teacher, but we see quite a big dip in performance in the teacher after DOGe fine-tuning. As you can see, the numbers are still quite far from the numbers reported in the paper. Are there other hyperparameters to tune, or could you explain why I'm not getting the exact results in the paper? Do you have any insights on how to reproduce the numbers in the paper?
